File: LeetcodeQuestions/leetcode_challenges/February_challange/score_parentheses.rb
def score_of_parentheses(s)
    stack = []
    score = 0
    s.each_char do |char|
        if char == "("
            stack << score
            score = 0
        else
            score = stack.pop + [score * 2, 1].max
        end
    end
    score
end

puts score_of_parentheses("()")
puts score_of_parentheses("(())")
puts score_of_parentheses("()()")
puts score_of_parentheses("(()(()))")

File: LeetcodeQuestions/leetcode_challenges/February_challange/score_parentheses.py
def scoreOfParentheses(S):
    score = 0
    stack = []

    for char in S:
        if char == "(":
            stack.append(score)
            score = 0
        else:
            score = stack.pop() + max(score * 2, 1)
    return score


print(scoreOfParentheses("()"))
print(scoreOfParentheses("(())"))
print(scoreOfParentheses("()()"))
print(scoreOfParentheses("(()(()))"))
